src/databricks/labs/ucx/source_code/dependency_containers.py

```diff
@@ -0,0 +1,28 @@
+from __future__ import annotations
+
+import abc
+import typing
+
+from databricks.labs.ucx.source_code.dependency_graph import DependencyGraph, Dependency
+
+if typing.TYPE_CHECKING:
```

Comment:
is it really necessary to add these `if typing.TYPE_CHECKING:` lines? it diverts the attention in the wrong direction, IMO